packages/client/src/BrubeckNode.ts

```diff
@@ -165,4 +165,30 @@ export default class BrubeckNode implements Context {
             this.debug('publishToNode << %o', streamMessage.getMessageID())
         }
     }
+
+    openPublishProxyConnectionOnStream(streamId: string, streamPartition: number | undefined, nodeId: string): void | Promise<void> {
+        const partition = streamPartition || 0
+        try {
+            if (!this.cachedNode || !this.startNodeComplete) {
+                return this.startNode().then((node) => {
+                    return node.joinStreamAsPurePublisher(streamId, partition, nodeId)
+                })
+            }
+            return this.cachedNode.joinStreamAsPurePublisher(streamId, partition, nodeId)
```

Comment:
```suggestion
            await this.cachedNode.joinStreamAsPurePublisher(streamId, partition, nodeId)
```

Reply:
The functions in the BrubeckNode class have a pattern of being able to be ran synchronously or asynchronously. This is why the function in question is returned instead of awaited.